src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -308,7 +308,7 @@ TxQ::canBeHeld(STTx const& tx, OpenView const& view,
             an early one fail or get dropped.
         */
         canBeHeld = accountIter == byAccount_.end() ||
-            !replacementIter ||
+            replacementIter ||
```

Comment:
So the old code was a bug? This code is checking if adding this transaction to the queue will overflow the max number of queued transactions. Since we're replacing a tx, that can never happen. If I'm wrong let me know and I'll re-review this change.